### PR TITLE
feat: first draft for GetInvitationsByUser

### LINKED ISSUE #193: Add `GetInvitationsByUser` method to organization-svc

### PR DESCRIPTION
**Pull request by florian-sabonchi**
closes #193

### CODE REVIEW

File: proto/services/user_svc/v1/organization_svc.proto [hunk 1]
```diff
@@ -148,6 +148,26 @@ message AcceptInvitationRequest {
 
 message AcceptInvitationResponse {}
 
+//
+// GetInvitationsByUser
+//
+
+message GetInvitationsByUserRequest {
+  string state = 2;
```
Comment (MaxSchaefer) [inline proto/services/user_svc/v1/organization_svc.proto:156]:
```suggestion
  optional string state = 1;
```

This change will break some logic. Cam we validate this using gotags?

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: services/user-svc/internal/organization/organization.go [hunk 2]
```diff
@@ -341,6 +342,45 @@ func (s ServiceServer) AcceptInvitation(ctx context.Context, req *pb.AcceptInvit
...
+	filter := db.Where("email = ?", claims.Email)
+	if req.State != nil {
+		filter = filter.Where("state = ?", req.State)
+	}
+
+	if err := filter.Find(&invitations).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			log.Warn().Err(err).Msg("database error")
+			return nil, status.Error(codes.Internal, err.Error())
+		}
```
Comment (FoseFx) [inline services/user-svc/internal/organization/organization.go:367]: else case is missing, please make sure this is deliberate, maybe add a comment

**Review by FoseFx — APPROVED** [review]

File: services/user-svc/internal/organization/organization.go [hunk 3]
```diff
@@ -341,6 +342,47 @@ func (s ServiceServer) AcceptInvitation(ctx context.Context, req *pb.AcceptInvit
 	return &pb.AcceptInvitationResponse{}, nil
 }
 
+func (s ServiceServer) GetInvitationsByUser(ctx context.Context, req *pb.GetInvitationsByUserRequest) (*pb.GetInvitationsByUserResponse, error) {
+
```
Comment (MaxSchaefer) [inline services/user-svc/internal/organization/organization.go:346]:
```suggestion
```

**Review by MaxSchaefer — APPROVED** [review]
Noice 🫡

**Pull request merged**